odm2api/ODM2/services/readService.py

```diff
@@ -694,13 +696,70 @@ def getDataSets(self, codes=None, uuids=None):
         if codes:
             q = q.filter(DataSets.DataSetCode.in_(codes))
         if uuids:
-            q.q.filter(DataSets.DataSetUUID.in_(uuids))
+            q.filter(DataSets.DataSetUUID.in_(uuids))
         try:
             return q.all()
         except Exception as e:
             print('Error running Query {}'.format(e))
             return None
 
+    def getSamplingFeatureDatasets(self, ids=None, codes=None, uuids=None, type=None):
```

Comment:
I see that `type` is used here as a variable, it is a python reserved word. Maybe as we continue to develop the api, we should try to follow better coding convention. So since this is a new function, `type` should be replaced by `dstype` or something along those lines to avoid using reserved python vocab.